tests/test_pytypes.py

```diff
@@ -220,6 +220,27 @@ def test_constructors():
         assert noconv2[k] is expected[k]
 
 
+def test_pybind11_str_raw_str():
+    # specifically to exercise pybind11::str::raw_str
+    cvt = m.convert_to_pybind11_str
+    assert cvt(u"Str") == u"Str"
+    assert cvt(b"Bytes") == u"Bytes" if str is bytes else "b'Bytes'"
+    assert cvt(None) == u"None"
+    assert cvt(False) == u"False"
+    assert cvt(True) == u"True"
+    assert cvt(42) == u"42"
+    assert cvt(2**65) == u"36893488147419103232"
+    assert cvt(-1.50) == u"-1.5"
+    assert cvt(()) == u"()"
+    assert cvt((18,)) == u"(18,)"
+    assert cvt([]) == u"[]"
+    assert cvt([28]) == u"[28]"
+    assert cvt({}) == u"{}"
+    assert cvt({3: 4}) == u"{3: 4}"
+    assert cvt(set()) == u"set([])" if str is bytes else "set()"
+    assert cvt({3, 3}) == u"set([3])" if str is bytes else "{3}"
```

Comment:
Another one, sorry. There're no funky unicode characters/string/weirdly encoded strings as bytes here!

Reply:
And while you're add it, I don't know how much work it is, but some checks on types with custom `__str__`/`__unicode__` methods might be interesting?